### PR TITLE
VPN-5187: Expose "Factory reset" functionality

### PR DESCRIPTION
**Pull request by MattLichtenstein**
## Description

- Exposes the "Reset and quit" functionality that was previously only available in the developer menu, to users via the "Get help" settings
- Resetting the VPN currently resets the following settings: 
  - All persistent settings stored in vpn.moz (list can be found in [settingslist.h](https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/src/settingslist.h))
  - iOS only: VPN tunnel system config
  - Authentication details. The user is signed out

## Reference

[VPN-5187: Implement new 'factory reset' UX in the GUI](https://mozilla-hub.atlassian.net/browse/VPN-5187)

### CODE REVIEW

File: src/ui/screens/settings/ViewReset.qml [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+            text: MZI18n.ResetSettingsBody1
+            color: MZTheme.theme.fontColor
+            horizontalAlignment: Text.AlignLeft
+        }
+
+        MZInterLabel {
+            Layout.topMargin: 16
+            Layout.fillWidth: true
+
+            text: "<ul style='margin-left: -20px;'><li>%1</li></ul>".arg(MZI18n.HelpSheetsPrivacyTitle)
```
Comment (mcleinman) [inline src/ui/screens/settings/ViewReset.qml:56]: Why are we styling text inline?

**Review by mcleinman — COMMENTED** [review]
Just one thing, for me to understand better. This is looking great.

File: src/ui/screens/settings/ViewReset.qml (hunk 1 above)
Comment (MattLichtenstein) [reply src/ui/screens/settings/ViewReset.qml:56]: It's markdown to get a bulleted list

Comment (mcleinman) [reply src/ui/screens/settings/ViewReset.qml:56]: Do we do similar things other places in the app? I don't see many. Is there any cleaner way to do a bulleted list? (Genuine question, I'm not familiar w/ Qt best practices.)

**Comment by mcleinman** [comment]
This factory reset option appears on the "Get help" menu on the initial log in screens. Is this intended? Presumably you're already in a reset state at that point.

File: src/ui/screens/settings/ViewReset.qml (hunk 1 above)
Comment (MattLichtenstein) [reply src/ui/screens/settings/ViewReset.qml:56]: I believe it's a standard practice. Alternatively we could build something ourselves like a rounded rectangle + text side by side, but I'm not sure that is any more beneficial. We use the same technique in a few places like guides and in-app messages, but, afaik we don't have bulleted list anywhere else

**Comment by MattLichtenstein** [comment]
> This factory reset option appears on the "Get help" menu on the initial log in screens. Is this intended? Presumably you're already in a reset state at that point.

Good question! Spoke with Santiago, and he is fine with the option being accessible from any "Get help" access point. Makes sense because a previously logged in user would still have all of their settings intact (recent connections, privacy features, etc) and could want to dispose of them before logging back in

**Review by mcleinman — APPROVED** [review]
Thanks!

File: src/ui/screens/settings/ViewReset.qml (hunk 1 above)
Comment (lesleyjanenorton) [reply src/ui/screens/settings/ViewReset.qml:61]: Curious: Did you happen to try this with a screen reader? I'm wondering how Qt exposes the text value and if the markup could cause any grief there.

**Pull request merged**